rfd/0089-merge-webapps.md

```diff
@@ -0,0 +1,185 @@
+---
+authors: Jeff Pihach (jeff.pihach@goteleport.com)
+state: draft
+---
+
+# RFD 89 - Merge Webapps into Teleport
+
+## What
+
+- Merge the [webapps](https://github.com/gravitational/webapps) repo into the
+  [teleport](https://github.com/gravitational/teleport) repo.
+- Merge the [webapps.e](https://github.com/gravitational/webapps.e) repo into
+  the [teleport.e](https://github.com/gravitational/teleport.e) repo.
+
+## Why
+
+The version of the web UI and the Teleport are tightly coupled but the
+repositories were created separate to support both Gravity and Teleport. This
+introduced complexities when integrating the two projects.
+
+We have to separately clone and build the web UI and commit the webasset
+changes into the teleport repo whenever we want to represent an update. We also
+have to maintain multiple build systems and ops procedures for the web UI.
+
+Now that Gravity is no longer a product we develop, moving the web UI
+repositories into their respective Teleport repo will save us time and
+confusion.
+
+## Success Criteria
+
+Both [Teleport](https://github.com/gravitational/teleport) and
+[Teleport.e](https://github.com/gravitational/teleport.e) repos contain their
+respective UI counterparts and the build systems correctly build all projects.
+
+## Developer Experience
+
+### Day to day
+
+The Day to day development experience shouldn't change. The webapps project
```

Comment:
Today we have the ability to run `DEBUG=1 teleport start ...` in order to load webassets from disk instead of the ones embedded in the binary.

Let's not lose this capability. You'll have to make a change to Teleport (probably before the migration) to update where it looks for webassets. Please play around and see if you can get this working, then update the "Actions" section at the end of this document.

Reply:
After playing with this a bit I think the best way to do this is to remove the `/webassets` submodule but keep the folder as the output location of the on-demand build. This will keep the functionality of `DEBUG=1` intact without any changes to `teleport` itself. I've updated the action below accordingly.